main.nf

```diff
@@ -0,0 +1,684 @@
+    file '*' into squid_output
+
+    script:
+    def avail_mem = task.memory ? "--limitBAMsortRAM ${task.memory.toBytes() - 100000000}" : ''
+    """
+    STAR \\
+        --genomeDir ${index} \\
+        --sjdbGTFfile ${gtf} \\
+        --runThreadN ${task.cpus} \\
+        --readFilesIn ${reads[0]} ${reads[1]} \\
+        --twopassMode Basic \\
+        --chimOutType SeparateSAMold --chimSegmentMin 20 --chimJunctionOverhangMin 12 --alignSJDBoverhangMin 10 --outReadsUnmapped Fastx --outSAMstrandField intronMotif \\
+        --outSAMtype BAM Unsorted ${avail_mem} \\
+        --readFilesCommand zcat
+    samtools sort Aligned.out.bam > Aligned.out.sorted.bam
+    samtools view -Shb Chimeric.out.sam > Chimeric.out.bam
+    squid -b Aligned.out.sorted.bam -c Chimeric.out.bam -o fusions
+    AnnotateSQUIDOutput.py ${gtf} fusions_sv.txt fusions_annotated.txt
+    """
+}
+
+/*************************************************************
+ * Summarizing results from tools
+ ************************************************************/
+process summary {
+    tag "$name"
+    publishDir "${params.outdir}/Report-${name}", mode: 'copy'
+ 
+    when:
+    !params.test && (params.fusioncatcher || params.star_fusion || params.ericscript || params.pizzly || params.squid)
+    
+    input:
+    set val(name), file(reads) from read_files_summary
+    file fusioncatcher from fusioncatcher_fusions.ifEmpty('')
+    file star_fusion from star_fusion_fusions.ifEmpty('')
+    file ericscript from ericscript_fusions.ifEmpty('')
+    file pizzly from pizzly_fusions.ifEmpty('')
+    file squid from squid_fusions.ifEmpty('')
+
+    output:
+    file 'fusions.txt' into summary_fusions
+    file 'summary.yaml' into summary_fusions_mq
+    file '*.html' into report
+    
+    script:
+    """
+    transformer.py -i ${fusioncatcher} -t fusioncatcher
+    transformer.py -i ${star_fusion} -t star_fusion
+    transformer.py -i ${ericscript} -t ericscript
+    transformer.py -i ${pizzly} -t pizzly
+    transformer.py -i ${squid} -t squid
+    generate_report.py fusions.txt summary.yaml -s ${name} -o .
+    """
+}
+
+/*************************************************************
+ * Visualization
+ ************************************************************/
+
+/*
+ * Fusion Inspector
+ */
+process fusion_inspector {
+    tag "$name"
+    publishDir "${params.outdir}/tools/FusionInspector", mode: 'copy'
+
+    when:
+    (params.fusion_inspector && !params.singleEnd) || (params.fusion_inspector && params.test)
+
+    input:
+    set val(name), file(reads) from read_files_fusion_inspector
+    file reference from fusion_inspector_ref.collect()
+    file summary_fusions
+
+    output:
+    file '*' into fusion_inspector_output
+
+    script:
+    """
+    FusionInspector \\
+        --fusions ${summary_fusions} \\
+        --genome_lib ${reference} \\
+        --left_fq ${reads[0]} \\
+        --right_fq ${reads[1]} \\
+        --CPU ${task.cpus} \\
+        --out_dir . \\
```

Comment:
Easy way to make the output more specific - specify an output directory and then pull everything from there.

Reply:
Done